server/datastore/mysql/hosts_test.go

```diff
@@ -2404,3 +2411,332 @@ func testHostsAllPackStats(t *testing.T, ds *Datastore) {
+			SystemTime:         152,
+			UserTime:           182,
+			WallTime:           2,
+		},
+		{
+			ScheduledQueryName: globalSQuery4.Name,
+			ScheduledQueryID:   globalSQuery4.ID,
+			QueryName:          globalQuery.Name,
+			PackName:           globalPack.Name,
+			PackID:             globalPack.ID,
+			AverageMemory:      8003,
+			Denylisted:         false,
+			Executions:         167,
+			Interval:           30,
+			LastExecuted:       time.Unix(1620325194, 0).UTC(),
+			OutputSize:         1340,
+			SystemTime:         153,
+			UserTime:           183,
+			WallTime:           3,
+		},
+		{
+			ScheduledQueryName: globalSQuery5.Name,
+			ScheduledQueryID:   globalSQuery5.ID,
+			QueryName:          globalQuery.Name,
+			PackName:           globalPack.Name,
+			PackID:             globalPack.ID,
+			AverageMemory:      8003,
+			Denylisted:         false,
+			Executions:         167,
+			Interval:           30,
+			LastExecuted:       time.Unix(1620325194, 0).UTC(),
+			OutputSize:         1340,
+			SystemTime:         153,
+			UserTime:           183,
+			WallTime:           3,
+		},
+	}
+
+	// Reload the host and set the scheduled queries stats for the scheduled queries that apply.
+	// Plus we set schedule query stats for a query that does not apply (globalSQuery1)
+	// (This could happen if the target platform of a schedule query is changed after creation.)
+	stats := make([]fleet.ScheduledQueryStats, len(globalStats))
+	for i := range globalStats {
+		stats[i] = globalStats[i]
+	}
+	stats = append(stats, fleet.ScheduledQueryStats{
+		ScheduledQueryName: globalSQuery1.Name,
+		ScheduledQueryID:   globalSQuery1.ID,
+		QueryName:          globalQuery.Name,
+		PackName:           globalPack.Name,
+		PackID:             globalPack.ID,
+		AverageMemory:      8003,
+		Denylisted:         false,
+		Executions:         167,
+		Interval:           30,
+		LastExecuted:       time.Unix(1620325194, 0).UTC(),
+		OutputSize:         1340,
+		SystemTime:         153,
+		UserTime:           183,
+		WallTime:           3,
+	})
+	host, err := ds.Host(context.Background(), host1.ID)
+	require.NoError(t, err)
+	host.PackStats = []fleet.PackStats{
+		{PackID: globalPack.ID, PackName: globalPack.Name, QueryStats: stats},
+	}
+	require.NoError(t, ds.SaveHost(context.Background(), host))
+
+	// Host should only return scheduled query stats only for the scheduled queries
+	// scheduled to run on "darwin".
+	host, err = ds.Host(context.Background(), host.ID)
+	require.NoError(t, err)
+	packStats := host.PackStats
+	require.Len(t, packStats, 1)
+	require.Len(t, packStats[0].QueryStats, 4)
+	sort.Slice(packStats[0].QueryStats, func(i, j int) bool {
+		return packStats[0].QueryStats[i].ScheduledQueryID < packStats[0].QueryStats[j].ScheduledQueryID
+	})
+	sort.Slice(globalStats, func(i, j int) bool {
+		return globalStats[i].ScheduledQueryID < globalStats[j].ScheduledQueryID
+	})
+	require.ElementsMatch(t, packStats[0].QueryStats, globalStats)
```

Comment:
We should probably make sure to add a `host2` that's `rhel` or something like that and see that it gets the other stats, including the one that has not ran yet.

Reply:
Good idea!